File: solutions/0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        
        INT_MAX=2**31-1
        INT_MIN=-2**31

        
        sign= 1 if x >= 0 else -1
        x=abs(x)
        r_s=str(x)[::-1]
        r_i=int(r_s)
        ret=sign*r_i
        if ret < INT_MIN or ret > INT_MAX:
            return 0
        return ret

File: solutions/0007-reverse-integer/solution.cpp
class Solution {
public:
    int reverse(int x) {
        long d=0,ret=0;
        while(x!=0){
            d=x%10;
            ret=ret*10+d;
            x=x/10;
        }
        if(ret>INT_MAX || ret<INT_MIN){
            return 0;
        }
        return ret;
    }
};
